Demo/MozillaSocial-iOS/MoSoContent/Auth.swift

```diff
@@ -0,0 +1,138 @@
+            decoder.keyDecodingStrategy = .convertFromSnakeCase
+            return try decoder.decode(ClientEntity.self, from: data)
+        } catch {
+            print("error: ", error)
+            return nil
+        }
+    }
+
+    func signIn(client: ClientEntity) {
+        var authURL = signInURL(with: client.clientId)
+
+        let session = ASWebAuthenticationSession(url: authURL, callbackURLScheme: redirectScheme) { (url, error) in
+            guard error == nil else {
+                print("ERROR: \(String(describing: error))")
+                return
+            }
+
+            guard let url = url, let code = self.parseCode(from: url) else {
+                print("Invalid Code Return")
+                return
+            }
+
+            Task {
+                self.authToken = await self.fetchAuthToken(for: client, with: code)
+            }
+        }
+
+        session.presentationContextProvider = self
+        session.prefersEphemeralWebBrowserSession = true
+        session.start()
+    }
+
+    private func fetchAuthToken(for client: ClientEntity, with code: String) async -> Token? {
+        var urlRequest = URLRequest(url: authTokenURL(for: client, with: code))
+        urlRequest.httpMethod = "POST"
+        do{
+            let (data, _) = try await URLSession.shared.data(for: urlRequest)
+
+            let decoder = JSONDecoder()
+            decoder.keyDecodingStrategy = .convertFromSnakeCase
+
+            return try decoder.decode(Token.self, from: data)
+        } catch {
+            print("error: ", error)
+        }
+
+        return nil
+    }
+
+    // MARK: - URL Builders
+
+    private func authTokenURL(for client: ClientEntity, with code: String) -> URL {
+        guard var url = URL(string: baseURL + "/oauth/token") else {
+            fatalError()
+        }
+
+        url.append(queryItems:[URLQueryItem(name: "code", value: code),
+                               URLQueryItem(name: "grant_type", value: "authorization_code"),
+                               URLQueryItem(name: "redirect_uri", value: redirectURI),
+                               URLQueryItem(name: "client_secret", value: client.clientSecret),
+                               URLQueryItem(name: "client_id", value: client.clientId),
+                               URLQueryItem(name: "scope", value: "read+write+push+follow")])
+        return url
+    }
+
+    private func registerAppURL() -> URL {
+        guard var registerURL = URL(string: baseURL + "/api/v1/apps") else {
+            fatalError()
+        }
+
+        registerURL.append(queryItems: [URLQueryItem(name: "client_name", value: "MozillaSocialDemo"),
+                                        URLQueryItem(name: "redirect_uris", value: redirectURI),
+                                        URLQueryItem(name: "scopes", value: "read write push follow")])
+        return registerURL
+    }
+
+    private func signInURL(with clientId: String) -> URL {
```

Comment:
nit: if you want, according to Apple best practices, identifiers should preferably named explicitly (like `clientIdentifier`) or abbreviated with an uppercase suffix (like `clientID`)

Reply:
For this one I followed the naming scheme provided by the Mastodon API, just to avoid having several names for the same property.
Normally `clientID` would be my go-to